pyiceberg/table/__init__.py

```diff
@@ -3542,14 +3629,22 @@ def __init__(self, transaction: Transaction, io: FileIO, snapshot_properties: Di
         self._io = io
         self._snapshot_properties = snapshot_properties
 
-    def fast_append(self) -> FastAppendFiles:
+    def fast_append(self, commit_uuid: Optional[uuid.UUID] = None) -> FastAppendFiles:
```

Comment:
Do we need to expose the `commit_uuid` to the public API?

Reply:
How about:
```
        with append_method() as append_files:
            append_files.commit_uuid = append_snapshot_commit_uuid
```
Maybe not the prettiest solution, but I would not expect an user to set the UUID, so we should not expose it to them

Reply:
Sounds good! Just to double confirm, I will remove the commit_uuid in the exisitng overwrite() API and apply this 'set-after-initialization' approach there as well?

Reply:
Fixed! I also removed the commit_uuid in overwrite().